doc/source/whatsnew/v0.25.0.rst

```diff
@@ -305,7 +305,7 @@ Conversion
 ^^^^^^^^^^
 
 - Bug in :func:`DataFrame.astype()` when passing a dict of columns and types the `errors` parameter was ignored. (:issue:`25905`)
--
+- Bug in :class:`DataFrame` construction from dict, where the index would be sorted instead of using dict insertion order. (:issue:`24859`)
```

Comment:
pls make clear of how this affects 3.5 and 3.6+

let's move this to the other api changes section as its not a bug rather an api change.